### PR TITLE
Update variable declarations to use ES6 style as suggested in #29

### LINKED ISSUE #29: Standardize code using ES6 identifier types
### Expected Behaviour

Use `const` type to declare identifiers that should **not** change it's value. If there is no way to keep identifier immutable, use `let` type to declare identifiers that should change it's value and **avoid** using `var` type

This are not rules, but guidelines, use your own judgement to identify the best type for each scenario.

More clarification [here](https://github.com/savoirfairelinux/ringme.js/pull/28#issuecomment-334481198)
### Actual Behaviour
At the moment all identifiers are using `var`

### Steps to reproduce

Just check the source code :smile:

### PR DESCRIPTION
**Pull request by nabrahamson**
### Changes proposed in this pull request:

* Updated variable and function declarations to use let or const. 
* const was used in cases where a function was being declared or the variable
   appeared to be immutable.  
* let was used for all others.

### Status

- [ ] READY
- [ ] HOLD
- [X] WIP (Work-In-Progress)

### How to verify this change

Verify that the variable name changes are to your liking.  

### Additional notes

The test web page will not load properly until you have an ES6 compiler (Babel / Webpack) implemented for this project.

I would reccomend further work to change function declaration to use the
syntax:
  "function foo(arg1, arg2) { }"

And the main function should be exported by:
  "export default function Ringme() {
   ...
   }"

### CODE REVIEW

File: src/ringme.js [hunk 1]
```diff
@@ -177,15 +177,15 @@ var RingMe = new function() {
    * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
    * SOFTWARE.
    */
-  var _launchUri = function(uri, successCallback, noHandlerCallback, unknownCallback) {
```
Comment (samnegri) [inline src/ringme.js:180]: There's a syntax error here, it is missing the attribution and the `function` keyword. This should be something like
```
const _lauchUri = function(uri, successCallback, noHandlerCallback, unknownCallback) {
```

File: src/ringme.js [hunk 2]
```diff
@@ -72,19 +72,19 @@ var RingMe = new function() {
...
     else {
-      var ringUI = _createRingUI.apply(this);
+      const ringUI = _createRingUI.apply(this);
 
       container.appendChild(ringUI);
     }
   }
 
-  var _createRingUI = function() {
-    var ui;
```
Comment (samnegri) [inline src/ringme.js:87]: Why not use `const` here?

File: src/ringme.js [hunk 3]
```diff
@@ -95,20 +95,20 @@ var RingMe = new function() {
     return ui;
   }
 
-  var _createButtonImage = function(buttonImage) {
-    var img = document.createElement('img');
```
Comment (samnegri) [inline src/ringme.js:99]: Why not use `const` here?

File: src/ringme.js [hunk 4]
```diff
@@ -95,20 +95,20 @@ var RingMe = new function() {
...
 
     return img;
   }
 
-  var _createAnchor = function(href, child) {
-    var ringUIInstance = this;
+  const _createAnchor = function(href, child) {
+    const ringUIInstance = this;
 
-    var anchor = document.createElement('a');
```
Comment (samnegri) [inline src/ringme.js:110]: Why not use `const` here?

File: src/ringme.js [hunk 5]
```diff
@@ -132,9 +132,9 @@ var RingMe = new function() {
     return anchor;
   }
 
-  var _ringMeClickEventHandler = function () {
+  const _ringMeClickEventHandler = function() {
     if (!this.isRingSchemeSupported()) {
-      var redirect = confirm(
```
Comment (samnegri) [inline src/ringme.js:137]: Why not use `const` here?

**Review by samnegri — CHANGES_REQUESTED** [review]
I think maybe you misunderstood the use of `const` ou maybe I'm missing something here. `const` identifier means that you can assign the value of the variable only once, but you can change the values or contents within the variable. Consider checking [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const)

**Comment by ssirois** [comment]
I have manually merge #43 in order to add credit where it belonged in this team effort.

Thanks to @samnegri, @nabrahamson and @combmag for all the work that gave us the possibility to close this issue rapidly and efficiently.

I will close this PR that have been deprecated by this manual push.

image

**Pull request closed without merging**